packages/gui/src/util/offerBuilderDataToOffer.ts

```diff
@@ -58,10 +68,12 @@ export default async function offerBuilderDataToOffer(
   const walletIdsAndAmounts: Record<string, BigNumber> = {};
   const driverDict: Record<string, Driver> = {};
 
-  const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
+  if (!allowEmptyOfferColumn) {
+    const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
 
-  if (!hasOffer) {
-    throw new Error(t`Please specify at least one offered asset`);
+    if (!hasOffer) {
+      throw new Error(t`Please specify at least one offered asset`);
+    }
```

Comment:
I think one-sideded offer includes request-only offer.
How about writing like this?
```typescript
  const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
  const hasRequest = !!requestedXch.length || !!requestedTokens.length || !!requestedNfts.length;
  if(!hasOffer && !hasRequest){
    throw new Error(t`Offer is empty`);
  }

  if (!allowEmptyOfferColumn) {
    if (!hasOffer || !hasRequest) {
      throw new Error(t`Please specify at least one offering/requesting asset`);
    }
  }
```

Reply:
I checked with E-ming, and the intent is to only support one-sided offers where assets are being offered. It isn't currently planned to support a request-only offer.

Reply:
In that case, the code looks good to me